ang/civicase/activity/feed/directives/activity-month-nav.directive.js

```diff
@@ -219,6 +220,24 @@
      */
     function initWatchers () {
       $scope.$on('civicaseActivityFeed.query', feedQueryListener);
+      $scope.$on('civicase::activity-feed::show-activity-panel', function () {
+        toggleMonthNavVisibility(false);
+      });
+      $scope.$on('civicase::activity-feed::hide-activity-panel', function () {
+        toggleMonthNavVisibility(true);
+      });
+    }
+
+    /**
+     * Toggles the visiblity of month nav,
+     * when hideQuickNavWhenDetailsIsVisible is true
+     *
+     * @param {Boolean} isVisible
+     */
+    function toggleMonthNavVisibility (isVisible) {
+      if ($scope.hideQuickNavWhenDetailsIsVisible) {
```

Comment:
this property comes from the parent directive too? I find it a bit difficult to follow directives that depend on their parent to define a certain property. Is this done for performance or convenience?

Reply:
It was done for convenience only. But np, I have created isolated scope and passed the variable from its parent.